File: Capitulo-08/Algoritimos_Gulosos/Exercicios/Java/src/GreedCoinChange.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class GreedCoinChange {
    public static List<Integer> greedyCoinChange(int[] coins, int amount) {
        Arrays.sort(coins);
        List<Integer> result = new ArrayList<>();

        for (int i = coins.length - 1; i >= 0; i--) {
            while (amount >= coins[i]) {
                amount -= coins[i];
                result.add(coins[i]);
            }
        }

        return result;
    }

    public static void main(String[] args) throws Exception {
        int[] coins = { 25, 10, 5, 1 };
        int amount = 63;
        List<Integer> result = greedyCoinChange(coins, amount);
        System.out.println(result);
    }
}

File: Capitulo-08/Algoritimos_Gulosos/Exercicios/Typescript/greed_coin_change.js
function greedyCoinChange(coins, amount) {
    coins.sort(function (a, b) { return b - a; });
    var result = [];
    for (var _i = 0, coins_1 = coins; _i < coins_1.length; _i++) {
        var coin = coins_1[_i];
        while (amount >= coin) {
            amount -= coin;
            result.push(coin);
        }
    }
    return result;
}
var coins = [25, 10, 5, 1];
var amount = 63;
console.log(greedyCoinChange(coins, amount));
